frontend/components/InlineFeedbackPopover/Comment.tsx

```diff
@@ -0,0 +1,146 @@
+  const [updatingCommentBody, setUpdatingCommentBody] = useState<string>(comment.body)
+
+  const [updateComment, { loading }] = useUpdateCommentMutation({
+    onCompleted: () => {
+      onUpdateComment()
+      setUpdatingCommentBody('')
+    },
+  })
+
+  const updateExistingComment = () => {
+    updateComment({
+      variables: {
+        commentId: comment.id,
+        body: updatingCommentBody,
+      },
+    })
+    setIsEditMode(false)
+  }
+
+  const [deleteComment, { loading: deleteLoading }] = useDeleteCommentMutation({
+    onCompleted: () => {
+      // just refetches the post as in updateComment
+      onUpdateComment()
+    },
+  })
+
+  const deleteExistingComment = () => {
+    deleteComment({
+      variables: {
+        commentId: comment.id,
+      },
+    })
+  }
+
+  return (
+    <div className="comment">
+      <div className="author-block">
+        {comment.author.profileImage ? (
+          <img className="profile-image" src={comment.author.profileImage} alt="" />
+        ) : (
+          <BlankAvatarIcon size={20} />
+        )}
+        <span className="author-identifier">
+          {comment.author.name
+            ? `${comment.author.name} (@${comment.author.handle})`
+            : `@${comment.author.handle}`}
+        </span>
+      </div>
+      <div className="body-block">
+        {isEditMode ? (
+          <textarea
+            value={updatingCommentBody}
+            onChange={(e) => setUpdatingCommentBody(e.target.value)}
+          />
+        ) : (
+          <p>{comment.body}</p>
+        )}
+        {canEdit && !isEditMode && (
+          <div className="edit-block">
+            <Button size={ButtonSize.Small} onClick={() => setIsEditMode(true)}>
+              edit
+            </Button>
+            <Button
+              size={ButtonSize.Small}
+              variant={ButtonVariant.Destructive}
+              onClick={deleteExistingComment}
+              disabled={deleteLoading}
+            >
+              delete
+            </Button>
+          </div>
+        )}
+        {canEdit && isEditMode && (
+          <Button size={ButtonSize.Small} onClick={updateExistingComment} disabled={loading}>
+            save
+          </Button>
+        )}
+      </div>
+      <style jsx>{`
+        .author-block {
+          display: flex;
+          align-items: center;
+          font-weight: bold;
+          font-size: 0.75em;
+        }
+
+        .author-block span {
+          margin-left: 5px;
+        }
+
+        .profile-image {
+          width: 27px;
+          height: 27px;
+          object-fit: cover;
+        }
+
+        :global(.author-block svg) {
+          border-radius: 50%;
+          background-color: ${theme.colors.blueLight};
+        }
+
+        .body-block {
+          display: flex;
+          justify-content: space-between;
+        }
+
+        .edit-block {
+          display: flex;
+        }
+
+        :global(button:first-child) {
```

Comment:
When I did just `:global(X)` in other components, I could have sworn I saw the styles apply to other elements outside of the component I was styling. It might be safer to do `.edit-block :global(button:first-child) {` or give a className to `<Button>`  and just use that class directly (which would also help decrease CSS specificity)

Reply:
Sounds good to me! Take a look at the updated 😄